src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java

```diff
@@ -107,10 +106,95 @@ private static Schema inferSchema(final String fieldPath, final BsonValue bsonVa
       case MAX_KEY:
       case UNDEFINED:
       default:
-        return DEFAULT_INFER_SCHEMA_TYPE;
+        return Schema.OPTIONAL_STRING_SCHEMA;
     }
   }
 
+  private static Schema combineArrayValueSchema(
+      final Schema firstSchema, final Schema secondSchema) {
+    if (isSentinel(firstSchema)) {
+      return secondSchema;
+    } else if (isSentinel(secondSchema)) {
+      return firstSchema;
+    }
+
+    if (firstSchema.equals(secondSchema)) {
+      return firstSchema;
+    }
+
+    if (firstSchema.type() != secondSchema.type()) {
+      LOGGER.debug(
+          "Can't combine non-matching schema types: {} and {}",
+          firstSchema.type(),
+          secondSchema.type());
+      return INCOMPATIBLE_SCHEMA_TYPE;
+    }
+
+    if (firstSchema.type() != Schema.Type.STRUCT || secondSchema.type() != Schema.Type.STRUCT) {
+      LOGGER.debug("Can't combine non-equal schema that are not both structs");
+      return INCOMPATIBLE_SCHEMA_TYPE;
+    }
+
+    SchemaBuilder builder = SchemaBuilder.struct().name(firstSchema.name()).optional();
+
+    // _id field first
```

Comment:
@jyemin Do you think this is needed for nested structs within arrays?

Reply:
There is no reason for it in nested structs.  There is likely no need for it anywhere.

Reply:
But seems like you can't remove it here without removing it everywhere, otherwise the order will change depending on whether this is one or more than one nested document in the array.

Reply:
Removed it in favour to natural sort order.

Reply:
Is there any context where this could be considered a backwards breaking change?

Reply:
Low risk but for example the following document would produce different field orders pre / post change: `{_a: 1, _id: 1}`

Reply:
Happy to revert c6c69a7 then there would be no backwards incompatibility.

Reply:
Annoyingly the `_` character is also after all the capital letters.  I think we should revert just to be safer.

Reply:
Ack